packages/holocron-dev-server/src/webpack/configs/loaders.js

```diff
@@ -0,0 +1,271 @@
+/*
+ * Copyright 2020 American Express Travel Related Services Company, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import TerserPlugin from 'terser-webpack-plugin';
+import ReactRefreshWebpackPlugin from '@pmmmwh/react-refresh-webpack-plugin';
+import { ESBuildMinifyPlugin } from 'esbuild-loader';
+
+import {
+  modulesLibraryVarName,
+  getWebpackVersion,
+  assetModuleFilename,
+  nodeModulesPattern,
+  jsxTest,
+  fileTest,
+  cssTest,
+} from '../helpers';
+
+export const fileLoader = ({ test = fileTest } = {}) => {
+  const webpackVersion = getWebpackVersion();
+  const fragment = {
+    module: {
+      rules: [],
+    },
+  };
+
+  if (webpackVersion >= 5) {
```

Comment:
Why are we handling multiple webpack versions?

Reply:
This was meant to future-proof the transition to the next major version of webpack - it's a major dependency to the library and as we gradually move to the next version, this would work out of the box.

Reply:
It enables us to set `webpack` a peer dependency as well and have that brought in by consumers of the lib (while supporting both versions)